R/hcai-impute.R

```diff
@@ -144,6 +150,14 @@ hcai_impute <- function(recipe,
     if (nominal_method == "new_category") {
       recipe <- step_missing(recipe, all_nominal(), - all_outcomes())
     } else if (nominal_method == "bagimpute") {
+      if ("character" %in% map_chr(recipe$template, ~{
+        class(.x) %>% first()
+      }))
+        warning("`bagimpute` depends on another library that does not support",
+                " character columns yet. If `bagimpute` does not impute missing",
+                " values, please convert all character columns to factors. If ",
+                "`collapse_rare_factors` is TRUE, the values that are not ",
+                'imputed might be contained in "other".')
```

Comment:
Doesn't this also apply to knn impute? The messages should be consistent, I think. It would be awesome to advise a user about how to check their data. They'd need to set `collapse_rare_factors = FALSE`, prep data, then using `missingness`